custom/enikshay/nikshay_datamigration/tests/test_create_enikshay_cases.py

```diff
@@ -0,0 +1,218 @@
+from collections import OrderedDict
+from datetime import date
+
+from django.core.management import call_command
+from django.test import TestCase
+
+from casexml.apps.case.sharedmodels import CommCareCaseIndex
+from corehq.apps.domain.models import Domain
+from corehq.apps.locations.models import SQLLocation, LocationType
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from custom.enikshay.nikshay_datamigration.models import Followup, Outcome, PatientDetail
+
+
+class TestCreateEnikshayCases(TestCase):
+
+    def setUp(self):
+        super(TestCreateEnikshayCases, self).setUp()
+
+        patient_detail = PatientDetail.objects.create(
+            PregId='MH-ABD-05-16-0001',
+            scode='MA',
+            Dtocode='Middlesex',
+            Tbunitcode=1,
+            pname='A B C',
+            pgender='M',
+            page=18,
+            paadharno=867386000000,
+            paddress='Cambridge MA',
+            pmob='9987328695',
+            pregdate1=date(2016, 12, 13),
+            cname='Secondary name',
+            caddress='Secondary address',
+            cmob='123',
+            dcpulmunory='Y',
+            dotname='Bubble Bubbles',
+            dotmob='321',
+            dotpType=1,
+            PHI=2,
+            atbtreatment='',
+            Ptype=4,
+            pcategory=4,
+        )
+        Outcome.objects.create(
+            PatientId=patient_detail,
+            HIVStatus='negative',
+        )
+        # Household.objects.create(
+        #     PatientID=patient_detail,
+        # )
+        for i in range(5):
+            Followup.objects.create(
+                id=(i + 1),
+                PatientID=patient_detail,
+            )
+
+        self.domain = Domain(name='enikshay-test-domain')
+        self.domain.save()
+
+        loc_type = LocationType.objects.create(
+            domain=self.domain.name,
+            name='nik'
+        )
+
+        SQLLocation.objects.create(
+            domain=self.domain.name,
+            location_type=loc_type,
+            metadata={
+                'nikshay_code': 'MH-ABD-05-16',
+            },
+        )
+
+        self.case_accessor = CaseAccessors(self.domain.name)
+
+    def tearDown(self):
+        Outcome.objects.all().delete()
+        Followup.objects.all().delete()
+        # Household.objects.all().delete()
+        PatientDetail.objects.all().delete()
+
+        self.domain.delete()
+
+        SQLLocation.objects.all().delete()
+        LocationType.objects.all().delete()
+
+        super(TestCreateEnikshayCases, self).tearDown()
+
+    def test_case_properties(self):
```

Comment:
I think it'd be good to have a test case for the matching workflow that makes sure we update in place and don't create new cases.